### DISCUSSION TITLE
Update_sa and Flush_sa are not working

### DISCUSSION THREAD

**Discussion by bhaskarbk018** [strongswan/strongswan#1099] [Q&A] [answered]
Hi Team ,  
    we are unable to call the update_sa and flush_sa in strongswan tried with lots of changes in configuration but its not working could you please help me on this and I am getting error like below . and unable to update new IKE_SA or delete the duplicate SA. 
failed to establish CHILD_SA, keeping IKE_SA
Jun 15 13:39:23 15[ENC] <london-to-vienna|8> generating CREATE_CHILD_SA response 166 [ N(TS_UNACCEPT) ]
Jun 15 13:39:23 15[NET] <london-to-vienna|8> sending packet: from 150.150.150.170[4500] to 150.150.150.179[4500] (80 bytes)
Jun 15 13:39:31 15[NET] <london-to-vienna|8> received packet: from 150.150.150.179[4500] to 150.150.150.170[4500] (208 bytes)
Jun 15 13:39:31 15[ENC] <london-to-vienna|8> parsed CREATE_CHILD_SA request 167 [ SA No TSi TSr ]
Jun 15 13:39:31 15[IKE] <london-to-vienna|8> traffic selectors 30.30.30.0/24 === 20.20.20.0/24 unacceptable
Jun 15 13:39:31 15[IKE] <london-to-vienna|8> failed to establish CHILD_SA, keeping IKE_SA

![image](https://user-images.githubusercontent.com/104621785/174273026-064a379b-0d06-449a-b731-5d7eb842a1bb.png)

Thanks in advance

**Comment by tobiasbrunner** [comment]
I don't see what the log has to do with `update_sa` or `flush_sas` (the latter is actually never used by strongSwan anymore), or why you'd call these manually. You seem to have a configuration issue (proposed and configured traffic selectors don't match).

**Comment by bhaskarbk018** [comment]
Thanks for the response. actually we need to know when this update_sa and flush_sas called in our strongswan we made so much change to call this upate_sa and flush_sas so could you please guide me on this. We need to know about this update_sa and flush_sas 
________________________________

**Reply by tobiasbrunner** [reply]
Sorry, I don't understand what exactly you want to know (call sites of these functions you can easily find by searching the code, but again, you won't find any for `flush_sas()` as that's never called).

**Comment by bhaskarbk018** [comment]
Thanks, we understand that flush_sa and flush_sp are not supported in strongswan; however we need sample ipsec.conf configuration that will trigger update_sa or a set of conditions which when simulated will trigger update_sa for strongswan. Much thanks for the help in advance.

**Comment by bhaskarbk018** [comment]
Thanks, we understand that flush_sa and flush_sp are not supported in strongswan; however we need sample ipsec.conf configuration that will trigger update_sa or a set of conditions which when simulated will trigger update_sa for strongswan. Much thanks for the help in advance.

**Reply by tobiasbrunner** [reply]
It's triggered by address/port changes of one or both of the peers e.g. due to a MOBIKE update (see the [ikev2/mobike-virtual-ip-nat](https://www.strongswan.org/testing/testresults/ikev2/mobike-virtual-ip-nat/index.html) test scenario for an example).

**Reply by deeptishm** [reply]
Hi, 
Under what conditions can the destination ip change be simulated for update_sa? 
Using https://www.strongswan.org/testing/testresults/ikev2/mobike-virtual-ip-nat/index.html, we could simulate change in src ip?

Thanks.

**Comment by strongX509** [comment, marked as answer]
This query is a duplicate to #1207 so I'm closing this discussion.
